lib/models/application.js

```diff
@@ -0,0 +1,1618 @@
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.application.pinToRelease('MyApp', 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	 * 	...
+	 * });
+	 *
+	 * @example
+	 * balena.models.application.pinToRelease(123, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	 * 	...
+	 * });
+	 *
+	 * @example
+	 * balena.models.application.pinToRelease('MyApp', 'f7caf4ff80114deeaefb7ab4447ad9c661c50847', function(error) {
+	 * 	if (error) throw error;
+	 * 	...
+	 * });
+	 */
+	exports.pinToRelease = (nameOrSlugOrId, fullReleaseHash, callback) =>
+		getId(nameOrSlugOrId)
+			.then(applicationId =>
+				pine.patch({
+					resource: 'application',
+					id: applicationId,
+					body: {
+						commit: fullReleaseHash,
+						should_track_latest_release: false,
+					},
+				}),
+			)
+			.asCallback(callback);
+
+	/**
+	 * @summary Get the hash of the current release for a specific application
+	 * @name getTargetReleaseHash
+	 * @public
+	 * @function
+	 * @memberof balena.models.application
+	 *
+	 * @param {String|Number} nameOrSlugOrId - application name (string), slug (string) or id (number)
+	 * @fulfil {String} - The release hash of the current release
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.application.getTargetReleaseHash('MyApp').then(function(release) {
+	 * 	console.log(release);
+	 * });
+	 *
+	 * @example
+	 * balena.models.application.getTargetReleaseHash(123).then(function(release) {
+	 * 	console.log(release);
+	 * });
+	 *
+	 * @example
+	 * balena.models.application.getTargetReleaseHash('MyApp', function(release) {
+	 * 	console.log(release);
+	 * });
+	 */
+	exports.getTargetReleaseHash = (nameOrSlugOrId, callback) =>
+		exports
+			.get(nameOrSlugOrId, { $select: 'commit' })
+			.get('commit')
+			.asCallback(callback);
+
+	/**
+	 * @summary Configure a specific application to track the latest available release
+	 * @name trackLatestRelease
+	 * @public
+	 * @function
+	 * @memberof balena.models.application
+	 *
+	 * @description The application's current release will be updated with each new successfully built release.
+	 *
+	 * @param {String|Number} nameOrSlugOrId - application name (string), slug (string) or id (number)
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.application.trackLatestRelease('MyApp').then(function() {
+	 * 	...
+	 * });
+	 *
+	 * @example
+	 * balena.models.application.trackLatestRelease(123).then(function() {
+	 * 	...
+	 * });
+	 *
+	 * @example
+	 * balena.models.application.trackLatestRelease('MyApp', function(error) {
+	 * 	if (error) throw error;
+	 * 	...
+	 * });
+	 */
+	exports.trackLatestRelease = (nameOrSlugOrId, callback) =>
+		releaseModel()
+			.getLatestByApplication(nameOrSlugOrId, {
+				$select: ['commit', 'belongs_to__application'],
+			})
+			.then(function(latestRelease) {
+				if (!latestRelease) {
+					return Promise.props({
+						applicationId: getId(nameOrSlugOrId),
+						/** @type {?string} */
```

Comment:
I had to add this as typescript with js will just default to using the type of the first return for every return and `null` was incompatible with the `string` below

Reply:
Not sure why this caused the docs to change, but changing to the following stopped it:
```suggestion
						/** @private @type {?string} */
```

Reply:
Done